### PR TITLE
Refactored render handler for api. The requests now fail per-target.

### PR DESCRIPTION
**Pull request by grzkv**
…e.g. if there are 3 targets and one fails, we still return a success with the data for 2 successful targets. Error handling is changed accordingly. Separate targets fail if at least one request fails with an error different from not-found.

### CODE REVIEW

File: app/carbonapi/http_handlers_test.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	"net/http"
 	"testing"
 
-	"github.com/bookingcom/carbonapi/pkg/types"
+	typ "github.com/bookingcom/carbonapi/pkg/types"
```
Comment (koorgoo) [inline app/carbonapi/http_handlers_test.go]: Should we change a reference not within this commit?
I see no conflicts in changes. Let's keep the diff as small as possible.

Comment (grzkv) [reply app/carbonapi/http_handlers_test.go]: This is an ongoing change. In many places, little by little. This is just one step of many. So, let's keep it.

File: app/carbonapi/http_handlers.go [hunk 2]
```diff
@@ -521,7 +526,7 @@ func (app *App) renderWriteBody(results []*types.MetricData, form renderForm, r
 	return body, nil
 }
 
-func errsFanIn(errs []error, n int) error {
+func optimistFanIn(errs []error, n int) error {
```
Comment (koorgoo) [inline app/carbonapi/http_handlers.go]: I suggest to change the func's signature to:
```
func expectNotFound(n int, errs []error) error {
  if len(errs) < n {
    return nil
  } 
  // continue without switch
}
```

File: app/carbonapi/http_handlers_test.go (hunk 1 above)
Comment (koorgoo) [reply app/carbonapi/http_handlers_test.go]: Ok then.

**Pull request merged**